src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -908,96 +911,148 @@ private List<Connection> getCongestedConnections() {
      * Assigns a list of nodes to each connection and fix net routes if there are cycles and / or multi-driver nodes.
      */
     protected void postRouteProcess() {
-        if (routeIteration <= config.getMaxIterations()) {
-            // perform LUT pin mapping updates
-            if (lutPinSwapping &&
-                    !Boolean.getBoolean("rapidwright.rwroute.lutPinSwapping.deferIntraSiteRoutingUpdates")) {
-                Map<SitePinInst, String> pinSwaps = new HashMap<>();
-                for (Connection connection: indirectConnections) {
-                    SitePinInst oldSinkSpi = connection.getSink();
-                    if (!oldSinkSpi.isLUTInputPin() || !oldSinkSpi.isRouted()) {
+        if (routeIteration > config.getMaxIterations()) {
+            return;
+        }
+
+        // perform LUT pin mapping updates
+        if (lutPinSwapping &&
+                !Boolean.getBoolean("rapidwright.rwroute.lutPinSwapping.deferIntraSiteRoutingUpdates")) {
+            Map<SitePinInst, String> pinSwaps = new HashMap<>();
+            for (Connection connection: indirectConnections) {
+                SitePinInst oldSinkSpi = connection.getSink();
+                if (!oldSinkSpi.isLUTInputPin() || !oldSinkSpi.isRouted()) {
+                    continue;
+                }
+
+                List<RouteNode> rnodes = connection.getRnodes();
+                RouteNode newSinkRnode = rnodes.get(0);
+                if (newSinkRnode == connection.getSinkRnode()) {
+                    continue;
+                }
+                connection.setSinkRnode(newSinkRnode);
+
+                SitePin newSitePin = newSinkRnode.getNode().getSitePin();
+                String existing = pinSwaps.put(oldSinkSpi, newSitePin.getPinName());
+                assert(existing == null);
+            }
+            LUTTools.swapMultipleLutPins(pinSwaps);
+        }
+
+        if (lutRoutethru) {
+            // When LUT routethru-s are considered, examine both static nets to find
+            // any cases where the *MUX output pin is used as a static source alongside
+            // the *_O output being used as a routethru. In such cases, configure the
+            // OUTMUX* site PIP to source from the 5LUT rather than the default 6LUT
+            // so that no conflict occurs
```

Comment:
Can you document the motivation for why we want to normalize on 6LUTs for routethrus and 5LUT for static nets?

Reply:
Comment expanded.